indra/newview/lllandmarkactions.cpp

```diff
@@ -265,6 +267,63 @@ void LLLandmarkActions::createLandmarkHere()
     createLandmarkHere(landmark_name, landmark_desc, folder_id);
 }
 
+void LLLandmarkActions::showFloaterCreateLandmarkForUrl(const std::string& url, const std::string& title)
+{
+    LLSLURL slurl(url);
+    if ((slurl.getType() != LLSLURL::LOCATION) &&
+        (slurl.getType() != LLSLURL::APP || slurl.getAppCmd() != LLSLURL::SLURL_REGION_PATH))
+    {
+        LL_INFOS() << "Unsupported URL: '" << url << "'" << LL_ENDL;
+        return;
+    }
+
+    LLSD data;
+    data["region"] = slurl.getRegion();
+    data["x"] = (S32)std::round(slurl.getPosition()[VX]);
+    data["y"] = (S32)std::round(slurl.getPosition()[VY]);
+    data["z"] = (S32)std::round(slurl.getPosition()[VZ]);
+    data["title"] = title.empty() ? slurl.getRegion() : title;
```

Comment:
May be use coordinates as a part of a tittle if you are using region's name? I suggest discussing this behavior wit vir. I understad that title might not be unavailable yet, but may be server can fill it in? create_inventory_landmark will be done anyway so might be a good chance to grab a tittle.